src/ServiceControl/Recoverability/Retrying/Infrastructure/ReturnToSenderDequeuer.cs

```diff
@@ -206,6 +184,10 @@ public virtual void Run(Predicate<TransportMessage> filter, CancellationToken ca
             {
                 Log.DebugFormat("Waiting for finish");
                 resetEvent.Wait(cancellationToken);
```

Comment:
Ideally, this should be non-blocking but you can argue that right now it is not a big deal because it is done in a continuation of endpoint start which is naturally scheduled and hopefully already away from the thread that entered the call context here